pkg/operator/ceph/cluster/status/conditions.go

```diff
@@ -0,0 +1,300 @@
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionIgnored(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionIgnored,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionUpgrading(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionReady,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionUpgrading,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionFailure(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionReady,
+		Status:  v1.ConditionFalse,
+		Reason:  "ClusterNotReady",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionFailure,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionProgressing(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionTrue,
+		Reason:  "",
+		Message: message,
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionProgressing,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionReady(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: message,
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionUpdating(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
```

Comment:
I see a lot of duplication, can't we combine using a single `Condition` exported function?

Reply:
I'll try to do that now.